cms/private_media/views.py

```diff
@@ -0,0 +1,34 @@
+from typing import TYPE_CHECKING
+
+from django.core.exceptions import PermissionDenied
+from django.shortcuts import redirect
+from wagtail.images.permissions import permission_policy
+from wagtail.images.views.serve import ServeView
+
+if TYPE_CHECKING:
+    from django.http import HttpResponse
+
+    from cms.private_media.models import AbstractPrivateRendition, PrivateImageMixin
+
+
+class PrivateImageServeView(ServeView):
+    def serve(self, rendition: "AbstractPrivateRendition") -> "HttpResponse":
+        image: PrivateImageMixin = rendition.image
+
+        # If there's no reason (within our control) for the file not to be served by
+        # media infrastructure, redirect
+        if image.is_public and not image.file_permissions_are_outdated():
+            return redirect(rendition.file.url)
+
+        # Block access to private images if the user has insufficient permissions
+        user = self.request.user
+        if image.is_private and (
+            not user.is_authenticated
+            or not permission_policy.user_has_any_permission_for_instance(user, ["choose", "add", "change"], image)
+        ):
+            raise PermissionDenied
```

Comment:
We need to ensure that the file can still be served if the image is private but has public pages referencing it. For example, if a page is published but the ACLs are not properly set due to network issues etc, the application should still serve the image until the next scheduled task attempts to fix and retry the ACL settings. We cannot have a page that features broken assets. The same principle applies to documents.

Reply:
This is covered by signal handlers. If a page features any media that is private, it is made public when those changes are published, and whilst the page remains published, media referenced by it will always remain public. 

Also, the `_privacy` field should always be the canonical indicator of privacy for a media item. If we're having to make additional queries at the time of serving to figure out if that is really the case, then we've failed terribly.

Reply:
>  if a page is published but the ACLs are not properly set due to network issues etc, the application should still serve the image until the next scheduled task attempts to fix and retry the ACL settings.

This particular case is handled by the last line of the method.